src/thor/triplegbuilder.cc

```diff
@@ -481,6 +482,72 @@ void SetHeadings(TripLeg_Edge* trip_edge,
   }
 }
 
+// TODO: how to use controller? or we don't need it at all?
+/**
+ * Add landmarks in directed edge to trip edge.
+ * @param  reader      Graph reader to get access to the graph tile and edge info.
+ * @param  trip_edge   Trip path edge to add landmarks.
+ * @param  controller  ?
+ * @param  edge        Directed edge where the landmarks are stored.
+ * @param  shape       Trip shape.
+ * @param  begin_index ?
+ */
+void AddLandmarks(GraphReader& reader,
+                  TripLeg_Edge* trip_edge,
+                  const AttributesController& controller,
```

Comment:
ok so the attributes controller is like a way in the request where the user can decide which metadata in the graph will come out the other side, especially for the purposes of map matching. in practice though we use the controller even for regular routes, for compatibility with things like osrms annotations. so yes we should in fact implement this but the good news is its pretty simple:

- add a new constant that controls this edge property: https://github.com/valhalla/valhalla/blob/97cad0609c2085b98196286b6cb02b0e09128749/valhalla/baldr/attributes_controller.h#L11
- make sure it is on by default: https://github.com/valhalla/valhalla/blob/97cad0609c2085b98196286b6cb02b0e09128749/src/baldr/attributes_controller.cc#L16
- add an if here in this function that actually checks if its enabled: https://github.com/valhalla/valhalla/blob/97cad0609c2085b98196286b6cb02b0e09128749/src/thor/triplegbuilder.cc#L823
- add docs to make sure people know it exists: https://github.com/valhalla/valhalla/blob/master/docs/docs/api/map-matching/api-reference.md#edge-items and in the master list above on that doc